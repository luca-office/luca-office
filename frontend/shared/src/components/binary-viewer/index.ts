export * from "./binary-viewer"
export * from "./image-view"
export * from "./video-view"
