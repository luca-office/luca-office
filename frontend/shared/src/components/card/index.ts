export * from "./card"
export * from "./elements"
export * from "./labelled-card"
export * from "./overview-card"
