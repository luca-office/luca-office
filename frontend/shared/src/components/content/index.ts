export * from "./content/content"
export * from "./content-bottom-actionbar/content-bottom-actionbar"
export * from "./content-missing-indicator/content-missing-indicator"
