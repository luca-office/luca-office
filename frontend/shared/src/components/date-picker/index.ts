export * from "./single-date-picker/single-date-picker"
