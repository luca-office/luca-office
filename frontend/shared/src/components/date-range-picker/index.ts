export * from "./date-range-picker"
export * from "./styles"
