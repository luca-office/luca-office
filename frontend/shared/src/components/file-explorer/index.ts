export * from "./file-explorer"
export * from "./tree-node"
