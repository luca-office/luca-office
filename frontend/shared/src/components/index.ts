export * from "./accordion-card/accordion-card"
export * from "./app-header/app-header"
export * from "./app-notification/app-notification"
export * from "./auth-info-column/auth-info-column"
export * from "./auth-info-column/auth-info-column-container"
export * from "./automated-coding-item-rule-field/automated-coding-item-rule-field"
export * from "./binary-upload-input/binary-upload-input"
export * from "./binary-viewer"
export * from "./book-table-of-contents/book-table-of-contents"
export * from "./breadcrumb/breadcrumb"
export * from "./button/button"
export * from "./card"
export * from "./chat"
export * from "./checkbox/checkbox"
export * from "./columns/columns"
export * from "./content"
export * from "./content-image/content-image"
export * from "./content-loading-indicator/content-loading-indicator"
export * from "./content-video/content-video"
export * from "./create-main-dimension-modal/create-main-dimension-modal"
export * from "./date-picker"
export * from "./date-range-picker"
export * from "./delete-or-archive-entity-button/delete-or-archive-entity-button"
export * from "./desktop/desktop"
export * from "./desktop/desktop-background"
export * from "./desktop/desktop-background-overlay"
export * from "./detail-view-card/detail-view-card"
export * from "./detail-view-header/detail-view-header"
export * from "./error-boundary/error-boundary"
export * from "./event-questionnaire/event-questionnaire"
export * from "./feature-disabled-marker/feature-disabled-marker"
export * from "./file-drop-zone/file-drop-zone"
export * from "./file-explorer"
export * from "./files-and-directories-detail-empty/files-and-directories-detail-empty"
export * from "./files-and-directories-detail-view/files-and-directories-detail-view"
export * from "./finish-modal/finish-modal"
export * from "./form-error-text/form-error-text"
export * from "./form-field-label/form-field-label"
export * from "./header-carousel/header-carousel"
export * from "./header-carousel/header-carousel-container"
export * from "./header-content-container/header-content-container"
export * from "./header-footer-container/header-footer-container"
export * from "./icon/icon"
export * from "./image-viewer/image-viewer"
export * from "./input"
export * from "./label/label"
export * from "./loading-indicator/loading-indicator"
export * from "./local-file-chip/local-file-chip"
export * from "./markdown-editor/markdown-editor"
export * from "./markdown-renderer/markdown-renderer"
export * from "./markdown-text/markdown-text"
export * from "./media-file-preview/media-file-preview"
export * from "./modal/modal"
export * from "./module-progress-indicator/module-progress-indicator"
export * from "./module-start-overlay/module-start-overlay"
export * from "./office-window/office-window"
export * from "./orly/orly"
export * from "./orly-button/orly-button"
export * from "./overlay/overlay"
export * from "./paper/paper"
export * from "./participant-questionnaire-detail/participant-questionnaire-detail"
export * from "./participant-scenario-detail/participant-scenario-detail"
export * from "./participant-scenario-detail/participant-scenario-detail"
export * from "./participant-scenario-detail/questionnaire-results-overlay"
export * from "./pdf-viewer/pdf-view"
export * from "./pdf-viewer/pdf-viewer"
export * from "./progress-bar/progress-bar"
export * from "./progress-bar-column/progress-bar-column"
export * from "./progress-time-clock/progress-time-clock"
export * from "./questionnaire"
export * from "./radio-button/radio-button"
export * from "./rating"
export * from "./react-portal/portal-utils"
export * from "./react-portal/react-portal"
export * from "./readonly-action-field/readonly-action-field"
export * from "./reference-book-chapters"
export * from "./report-participant-overview-table/report-participant-overview-table"
export * from "./reporting-final-score"
export * from "./reporting-progress-info/reporting-progress-info"
export * from "./reporting-questionnaire-overlay/reporting-questionnaire-overlay"
export * from "./reporting-scenario-overlay/reporting-scenario-overlay"
export * from "./reset-password-link/reset-password-link"
export * from "./scenario-coding-model-table-of-contents/scenario-coding-model-table-of-contents"
export * from "./scenario-info-card/info-tag"
export * from "./scenario-info-card/scenario-info-card"
export * from "./scenario-info-card/scenario-info-card"
export * from "./select/custom-select"
export * from "./selectable-card/selectable-card"
export * from "./selection-in-preview-footer/selection-in-preview-footer"
export * from "./settings-footer-card/settings-footer-card"
export * from "./slide-menu/slide-menu"
export * from "./spreadsheet"
export * from "./spreadsheet-preview/spreadsheet-file-preview"
export * from "./sub-header/sub-header"
export * from "./tab-button-bar/tab-button-bar"
export * from "./table/table"
export * from "./table/table-container"
export * from "./table-of-content"
export * from "./table-of-content/table-of-contents-entry/table-of-contents-entry"
export * from "./table-of-content/table-of-contents-entry-text-placeholder/table-of-contents-entry-text-placeholder"
export * from "./tag/tag"
export * from "./tag-placeholder/tag-placeholder"
export * from "./taskbar/taskbar"
export * from "./text-document-preview/text-document-preview"
export * from "./textarea/textarea"
export * from "./tooltip/tooltip"
export * from "./typography/typography"
export * from "./update-text-document/update-text-document"
export * from "./upload-file-modal/upload-file-modal"
export * from "./video-viewer/video-viewer"
export * from "./viewer-tools"
export * from "./warning-tooltip/warning-tooltip"
export * from "./window-action-bar/window-action-bar"
