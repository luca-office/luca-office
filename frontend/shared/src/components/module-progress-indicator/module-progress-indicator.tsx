import {css} from "@emotion/react"
import * as React from "react"
import {Text} from "../../components"
import {ProgressIndicatorStatus} from "../../enums"
import {
  chartCompleteBarColor,
  chartIncompleteBarColor,
  gradientPrimary,
  spacingSmall,
  spacingTiny,
  TextSize
} from "../../styles"

export interface ProgressIndicatorProps {
  readonly progressEntities: ProgressIndicatorStatus[]
}

export const ModuleProgressIndicator: React.FC<ProgressIndicatorProps> = ({progressEntities}) => {
  const renderProgressIndicatorEntry = (index: number, status: ProgressIndicatorStatus) => (
    <div
      key={`project_module_${index}`}
      css={styles.indicator(status)}
      className={getClassNameByProgressIndicatorStatus(status)}
    />
  )

  return (
    <div css={styles.content}>
      <Text size={TextSize.Medium}>{`${progressEntities.filter(e => e === ProgressIndicatorStatus.Completed).length}/${
        progressEntities.length
      }`}</Text>
      <div css={styles.indicatorWrapper(progressEntities.length)}>
        {progressEntities.map((progress, index) => {
          return renderProgressIndicatorEntry(index, progress)
        })}
      </div>
    </div>
  )
}

export const getClassNameByProgressIndicatorStatus = (status: ProgressIndicatorStatus): string =>
  status === ProgressIndicatorStatus.Completed
    ? "completed-indicator"
    : status === ProgressIndicatorStatus.InProgress
    ? "in-progress-indicator"
    : "open-indicator"

const Sizes = {
  indicator: spacingSmall
}

const styles = {
  content: css({
    display: "grid",
    gridTemplateColumns: "minmax(min-content, max-content) 1fr",
    gridColumnGap: spacingSmall,
    alignItems: "center"
  }),
  indicatorWrapper: (projectModulesCount: number) =>
    css({
      display: "grid",
      gridTemplateColumns: `repeat(${projectModulesCount}, 1fr)`,
      gridColumnGap: spacingTiny
    }),
  indicator: (status: ProgressIndicatorStatus) =>
    css({
      height: Sizes.indicator,
      background:
        status === ProgressIndicatorStatus.Completed
          ? chartCompleteBarColor
          : status === ProgressIndicatorStatus.InProgress
          ? gradientPrimary
          : chartIncompleteBarColor
    })
}
