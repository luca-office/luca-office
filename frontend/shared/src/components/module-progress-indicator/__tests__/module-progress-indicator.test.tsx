import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Text} from "../../../components"
import {ProgressIndicatorStatus} from "../../../enums"
import {ModuleProgressIndicator, ProgressIndicatorProps} from "../module-progress-indicator"

const defaultProps: ProgressIndicatorProps = {
  progressEntities: [
    ProgressIndicatorStatus.Completed,
    ProgressIndicatorStatus.Completed,
    ProgressIndicatorStatus.Open,
    ProgressIndicatorStatus.Open,
    ProgressIndicatorStatus.Open
  ]
}

const getComponent = (props?: Partial<ProgressIndicatorProps>) => (
  <ModuleProgressIndicator {...{...defaultProps, ...props}} />
)

describe("rater-rating-participant-progress-indicator", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure", () => {
    const component = getComponent()
    const tree = shallow(component)

    expect(tree.find(Text)).toHaveLength(1)
    expect(tree.find(".completed-indicator")).toHaveLength(2)
    expect(tree.find(".open-indicator")).toHaveLength(3)
  })
})
