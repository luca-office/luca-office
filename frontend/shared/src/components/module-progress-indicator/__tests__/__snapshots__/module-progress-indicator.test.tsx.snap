// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rater-rating-participant-progress-indicator renders correctly 1`] = `
<div
  className="css-ij7jn1-ModuleProgressIndicator"
>
  <div
    className="css-17zbm7y-Text"
  >
    2/5
  </div>
  <div
    className="css-1t82cj8-ModuleProgressIndicator"
  >
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
    <div
      className="completed-indicator css-c39ufe-ModuleProgressIndicator"
    />
    <div
      className="open-indicator css-kbhrsj-ModuleProgressIndicator"
    />
    <div
      className="open-indicator css-kbhrsj-ModuleProgressIndicator"
    />
    <div
      className="open-indicator css-kbhrsj-ModuleProgressIndicator"
    />
  </div>
</div>
`;
